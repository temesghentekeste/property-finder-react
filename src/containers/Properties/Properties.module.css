.properties {
  margin: 1rem;
}

.properties__container {
  display: flex;
  overflow-x: auto;
  margin-top: 1rem;
}

.properties__heading {
  text-align: center;
}

.scrollHorizontally {
  text-align: center;
  font-weight: bold;
  color: #444;
}

@media (min-width: 768px) {
  .properties__container {
    flex-wrap: wrap;
    justify-content: center;
  }

  .scrollHorizontally {
    display: none;
  }
}
