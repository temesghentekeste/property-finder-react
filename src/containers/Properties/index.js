import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { WaveLoading } from 'react-loadingg';
import styles from './Properties.module.css';

import { getPropertiesAsync, createFavoriteAsync } from '../../redux/api/apiActions';

import PropertyItem from '../../components/PropertyItem';

const Properties = () => {
  const [favoriteFlag, setFavoriteFlag] = useState(false);
  const dispatch = useDispatch();
  const { loading, properties, error } = useSelector(
    (state) => state.properties,
  );

  useEffect(() => {
    const abortController = new AbortController();
    dispatch(getPropertiesAsync());
    return () => {
      abortController.abort();
    };
  }, [dispatch, favoriteFlag]);

  const handleFavorirtes = (propertyId, value) => {
    dispatch(createFavoriteAsync(propertyId));
    setFavoriteFlag(value);
  };

  if (error) {
    return <Redirect to="/" />;
  }

  if (loading || loading === null || loading === undefined) {
    return <WaveLoading />;
  }

  if (
    !properties
    || properties === undefined
    || properties.data === undefined
  ) {
    return <WaveLoading />;
  }

  if (properties && properties.error) {
    return <h1>Something went wrong, please try again!</h1>;
  }

  return (
    <>
      <section className={styles.properties}>
        <h1>All Properties</h1>
        <div className={styles.properties__container}>
          {properties.data.length > 0
            && properties.data.map((property) => (
              <PropertyItem
                key={property.id}
                id={property.id}
                attributes={property.attributes}
                handleFavorirtes={handleFavorirtes}
                favoriteFlag={favoriteFlag}
              />
            ))}
        </div>
      </section>
      <p className={styles.scrollHorizontally}>Scroll Horizontally</p>
    </>
  );
};

export default Properties;
