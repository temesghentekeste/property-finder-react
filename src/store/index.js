import { configureStore } from '@reduxjs/toolkit';
import propertiesReducer from '../redux/propertiesSlice';
import propertyReducer from '../redux/propertySlice';
import loginReducer from '../redux/loginSlice';
import favoritesReducer from '../redux/favoritesSlice';
import signupReducer from '../redux/signupSlice';

export default configureStore({
  reducer: {
    properties: propertiesReducer,
    property: propertyReducer,
    usernametoken: loginReducer,
    favorites: favoritesReducer,
    signupuser: signupReducer,
  },
});
