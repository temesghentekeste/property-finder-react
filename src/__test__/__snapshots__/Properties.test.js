import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import Properties from '../../containers/Properties';
import store from '../../store';

const properties = [
  {
    data: [
      {
        id: '59',
        type: 'property',
        attributes: {
          name: 'MMX',
          address: 'Nsambia, Texas Club',
          monthly_price: '500.0',
          description: 'bb',
          is_for_rent: true,
          featured_image:
            'https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg',
          is_favorite: false,
        },
      },
    ],
  },
];

it('renders the Properties component', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Properties properties={properties} />
          </Router>
          ,
        </Provider>,
      )
      .toJSON();
  
    expect(component).toMatchSnapshot();
  });
