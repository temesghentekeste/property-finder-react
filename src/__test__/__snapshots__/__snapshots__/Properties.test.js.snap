// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Properties component 1`] = `
Array [
  <div
    className="sc-gzVnrw lGDtb"
    size="default"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "margin": "auto",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      className="sc-htoDjs sc-dnqmqq SDTxR"
    />
    <div
      className="sc-htoDjs sc-iwsKbI kUYcSE"
    />
    <div
      className="sc-htoDjs sc-gZMcBi eqPTXf"
    />
    <div
      className="sc-htoDjs sc-gqjmRU iFOlOR"
    />
    <div
      className="sc-htoDjs sc-VigVT kWPrjt"
    />
  </div>,
  ",",
]
`;
